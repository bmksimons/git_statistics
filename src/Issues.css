
/* .issue {
    padding: 20px;
    border: 1px solid black;
    background-color: bisque;
    width: 100px;
    height: 200px;
} */

.issueList {
    display: grid;
    grid-template-columns: repeat( auto-fit, minmax(150px, 1fr));
    gap: 10px;
    justify-items: center;
}

.wrapper {
    margin: 5% 10%
}

#selectFilterIssue {
    margin: 15px;
    padding: 8px;
    border: 1px solid black;
    background-color: bisque;
    border-radius: 10px;
}





