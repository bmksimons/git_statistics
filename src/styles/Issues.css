.issueList {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(150px, 1fr));
    height: 100%;
    gap: 10px;
    justify-items: center;
}

.wrapper {
    margin: 5% 10%
}

#selectFilterIssue {
    margin: 15px;
    padding: 8px;
    border: 1px solid black;
    background-color: bisque;
    border-radius: 10px;
}





