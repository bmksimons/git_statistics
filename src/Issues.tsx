import { GITLAB_TOKEN } from "./tokens";
import "./Issues.css";
import React, { MouseEvent } from 'react';

export default function Issues() {
  let issues: string[] = ["issue 1sssssssssss sssssssssssssss", "issue 2", "issue 3", "issue 4", "issue 5", "issue 6", "issue 7"]
  let issueItems = issues.map((issue) =>
  <div className = "issue" onClick = {showIssueDescription}>{issue}</div>
  );
  return (
    <div className = "wrapper">
      <h1>Issues</h1>
      <div className = "issueList">
        {issueItems}
      </div>
    </div>
    
  );
  
function showIssueDescription(event: MouseEvent<HTMLDivElement>) {
  console.log(event);
  
}
}

