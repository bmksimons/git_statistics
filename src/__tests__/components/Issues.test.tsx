import renderer from 'react-test-renderer';
import {render, screen, cleanup} from '@testing-library/react'
import Issues from '../../components/Issues'
import userEvent from '@testing-library/user-event'

afterEach(cleanup);

test('renders issues', () => {
    const tree = renderer.create(<Issues />).toJSON();
    expect(tree).toMatchSnapshot();
  });

describe('testing filtering issues', () => {

  test('default filter is show all', () => {
      const {getByTestId} = render(<Issues/>);
      expect(getByTestId("selectFilterIssue")).toHaveTextContent("Show all");
  });

  test('filter changes to "Show closed"', () => {
    render(<Issues />);
    userEvent.selectOptions(screen.getByRole('combobox'), [screen.getByText("Show closed")]);
    const title = screen.getByRole('option', {name: "Show closed"}) as HTMLOptionElement;
    expect(title.selected).toBe(true);
  });

  test('filter changes to "Show open"', () => {
    render(<Issues />);
    userEvent.selectOptions(screen.getByRole('combobox'), [screen.getByText("Show open")]);
    const title = screen.getByRole('option', {name: "Show open"}) as HTMLOptionElement;
    expect(title.selected).toBe(true);
  });
  
});