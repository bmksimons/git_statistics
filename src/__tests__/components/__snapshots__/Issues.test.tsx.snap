// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders issues 1`] = `
<div
  className="wrapper"
>
  <h2
    className="component-header"
  >
    Issues
  </h2>
  <p
    className="component-description"
  >
    Overview over all issues from the project, with the ability to filter between showing all, opened or closed issues. An issue has an issuenumber, title and labels. If you click on an issue the description will show up in a pop-up.
  </p>
  <div
    className="selectFilter"
  >
    <select
      data-testid="selectFilterIssue"
      id="selectFilterIssue"
      onChange={[Function]}
    >
      <option
        disabled={true}
        hidden={true}
        selected={true}
        value=""
      >
        
      </option>
      <option
        value="all"
      >
        Show all
      </option>
      <option
        value="opened"
      >
        Show open
      </option>
      <option
        value="closed"
      >
        Show closed
      </option>
    </select>
  </div>
  <div
    className="issueList"
  />
</div>
`;
